--- origsrc/kinit-5.27.0/src/kdeinit/kinit.cpp	2016-10-02 04:52:57.000000000 -0500
+++ src/kinit-5.27.0/src/kdeinit/kinit.cpp	2016-11-02 00:14:49.671940400 -0500
@@ -95,6 +95,10 @@ static const char *extra_libs[] = {
     "libKF5KIOCore.5.dylib",
     "libKF5Parts.5.dylib",
     "libKF5Plasma.5.dylib"
+#elif defined(Q_OS_CYGWIN)
+    "cygKF5KIOCore-5.dll",
+    "cygKF5Parts-5.dll",
+    "cygKF5Plasma-5.dll"
 #else
     "libKF5KIOCore.so.5",
     "libKF5Parts.so.5",
@@ -461,7 +465,11 @@ static pid_t launch(int argc, const char
     if (_name[0] != '/') {
         name = _name;
         lib = QFile::decodeName(name);
+#ifdef Q_OS_CYGWIN
+        libpath = QLatin1String("cygkdeinit5_") + lib + QLatin1String(".dll");
+#else
         libpath = QLatin1String("libkdeinit5_") + lib;
+#endif
         libpath_relative = true;
         execpath = execpath_avoid_loops(name, envc, envs, avoid_loops);
     } else {
@@ -470,7 +478,7 @@ static pid_t launch(int argc, const char
         name = name.mid(name.lastIndexOf('/') + 1);
 
         // FIXME: this .so extension stuff is very Linux-specific
-        if (lib.endsWith(QLatin1String(".so"))) {
+        if (lib.endsWith(QLatin1String(".so")) || lib.endsWith(QLatin1String(".dll"))) {
             libpath = lib;
         } else {
             execpath = _name;
@@ -482,6 +490,15 @@ static pid_t launch(int argc, const char
             // the install prefix, although this may not be the case if the user
             // has overridden them, and so this search is inherently fragile in
             // the face of unusual installation layouts.
+#ifdef Q_OS_CYGWIN
+            if (lib.contains(QLatin1String(KF5_LIBEXEC_INSTALL_DIR))) {
+                libpath = QString(lib).replace(QLatin1String(KF5_LIBEXEC_INSTALL_DIR),
+                                               QLatin1String("/bin/cygkdeinit5_")) + QLatin1String(".dll");
+            } else if (lib.contains(QLatin1String("/bin/"))) {
+                libpath = QString(lib).replace(QLatin1String("/bin/"),
+                                               QLatin1String("/bin/cygkdeinit5_")) + QLatin1String(".dll");
+            }
+#else
             if (lib.contains(QLatin1String(KF5_LIBEXEC_INSTALL_DIR))) {
                 libpath = QString(lib).replace(QLatin1String(KF5_LIBEXEC_INSTALL_DIR),
                                                QLatin1String(LIB_INSTALL_DIR "/libkdeinit5_")) + QLatin1String(".so");
@@ -489,6 +506,7 @@ static pid_t launch(int argc, const char
                 libpath = QString(lib).replace(QLatin1String("/bin/"),
                                                QLatin1String(LIB_INSTALL_DIR "/libkdeinit5_")) + QLatin1String(".so");
             }
+#endif
             // Don't confuse the user with "Could not load libkdeinit5_foo.so" if it doesn't exist
             if (!QFile::exists(libpath)) {
                 libpath.clear();
